src/vm/eventpipeevent.cpp

```diff
@@ -25,6 +25,8 @@ EventPipeEvent::EventPipeEvent(EventPipeProvider &provider, INT64 keywords, unsi
     m_level = level;
     m_needStack = needStack;
     m_enabled = false;
+    m_pMetadata = pMetadata;
+    m_metadataLength = metadataLength;
```

Comment:
Per my comment above, this is where you want to allocate a new buffer and copy the buffer pointed to by pMetadata into it.  Then free the buffer in ~EventPipeEvent.